src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -24,18 +29,24 @@
     private HttpServletResponse response;
     @Mock
     private RequestDispatcher requestDispatcher;
-
+    @Mock
+    private ArrayListProductDao productDao;
+    @Spy
+    private List<Product> products;
+    @InjectMocks
     private ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() {
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+        when(productDao.findProducts()).thenReturn(products);
     }
 
     @Test
     public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
 
+        verify(productDao).findProducts();
```

Comment:
Verify that this list is attached to the request object, not just is being retrieved. Also add some mock products to it in setup method.